examples/lock-app/silabs/src/LockManager.cpp

```diff
@@ -475,18 +500,39 @@ bool LockManager::GetCredential(chip::EndpointId endpointId, uint16_t credential
     ChipLogProgress(Zcl, "Lock App: LockManager::GetCredential [credentialType=%u], credentialIndex=%d",
                     to_underlying(credentialType), credentialIndex);
 
-    const auto & credentialInStorage = mLockCredentials[to_underlying(credentialType)][credentialIndex];
+    uint16_t size = static_cast<uint16_t>(sizeof(LockCredentialInfo));
+
+    chip::StorageKeyName key = LockCredentialEndpoint(credentialIndex, credentialType, endpointId);
+
+    error = chip::Server::GetInstance().GetPersistentStorage().SyncGetKeyValue(key.KeyName(), &credentialInStorage, size);
```

Comment:
same comment as userInStorage, credentialInStorage doesn't appear to need to be a member of the LockManager as all its value are copied into the `& credential` reference passed to this method.